### PR TITLE
Fix all errors pointed out by the following yetus plugins: blanks

### PR DESCRIPTION
**Pull request by rene**
When running yetus locally through `make yetus` some plugins still point errors already present on master branch, which might confuse developers when adding new code, since it will require to them to open yetus' logs in order to check whether errors are coming from master or from the new code. This PR fixes all errors pointed out by the following plugins:

- blanks

Making them passing as shown by the following output from yetus:

```text
| Vote |        Subsystem |  Runtime   | Comment
============================================================================
|   0  |          reexec  |   0m 33s   | Precommit patch detected.
+---------------------------------------------------------------------------
|      |                  |            | Prechecks
+---------------------------------------------------------------------------
|  +1  |         dupname  |   7m 15s   | No case conflicting files found.
+---------------------------------------------------------------------------
|      |                  |            | Compile Tests
+---------------------------------------------------------------------------
|  -1  |       codespell  |   0m 38s   | The source tree has 39 issues.
|  -1  |      detsecrets  |  19m  0s   | The source tree has 59 issues.
|  +1  |         compile  |   0m 53s   | the source passed
|  +1  |              cc  |   0m 53s   | the source passed
|  +1  |          golang  |   0m 53s   | the source passed
|  +1  |          blanks  |   0m 17s   | The source tree has no blanks issues.
|  -1  |         buflint  |   0m 20s   | The source tree has 881 issues.
|  +1  |       checkmake  |   0m  0s   | No issues.
|  -1  |    golangcilint  |  10m 31s   | The source tree has 2990 issues.
|  +1  |        hadolint  |   0m  1s   | No issues.
|  +1  |        jsonlint  |   0m  3s   | No issues.
|  +1  |    markdownlint  |   1m 45s   | No issues.
|  +1  |          pylint  |   0m  7s   | No issues.
|  -1  |          revive  |   1m  8s   | The source tree has 572 issues.
|  -1  |      shellcheck  |  18m  6s   | The source tree has 357 issues.
|  +1  |         xmllint  |   0m  2s   | No issues.
|  +1  |        yamllint  |   0m 14s   | No issues.
+---------------------------------------------------------------------------
|      |                  |            | Other Tests
+---------------------------------------------------------------------------
|      |                  |  62m 14s   |
```

### CODE REVIEW

**Comment by rene** [comment]
FYI: @ruslan-zededa , @milan-zededa 

Please, add another developers related to the changed files.

File: docs/DEBUGGING.md
Comment (eriknordmark) [inline docs/DEBUGGING.md:91]: Maybe the langauge tag should be text instead of sh above. 
In anyn case, it doesn't make sense replacing the output of a command example just because blanks doesn't like it.

**Review by eriknordmark — COMMENTED** [review]
We've had cases in the past when applying the suggestions from the various tools used by yetus has resulted in subtle bugs. So we can't do that.

When yetus runs as part of a PR build it produces annotations which can be seen in the "files changed" tab in github.

I was about to suggest that blanks and codespell could be applied blindly, but not the other.
But then I see that blanks in this PR wants to correct verbatim output of a shell command, which makes no sense (even if it doesn't result in a bug AFAIK).
If codespell was limited to comments and markdown then it would be safe, but I've seen one case when it was commenting on a variable name ("ans" is apparently a mistyped "and" according to codespell) and I can see cases where renaming a variable can lead to subtle bugs (supposed the code has one variable called "answer" and another called "ans", and as a result of codespell the user renames "ans" to "answer" and not things are broken.)

So I think we need to be a lot more careful with doing bulk changes based on these tools.
Maybe doing a PR with only blanks (and even in that case not doing them blindly), and a separate one based on codespell?

But in general we will have complains creep into existing code since the defaults for the tools sometimes changes when there are new releases of these tools.

File: api/proto/info/info.proto
Comment (giggsoff) [inline api/proto/info/info.proto:445]: Will proto updates produce generated code update? If so, nice to run `make proto&&make proto-vendor` and provide updates in separate commit.

**Comment by rene** [comment]
> We've had cases in the past when applying the suggestions from the various tools used by yetus has resulted in subtle bugs. So we can't do that.
> 

I agree, and that's why I've requested more developers related to the changed files to be added for review. 

> When yetus runs as part of a PR build it produces annotations which can be seen in the "files changed" tab in github.
> 

For PR's it's OK, we clear can see when yetus passed or not because of our changes. My main concern is when running yetus locally, I really don't want to see those big FAILURE message results since it wasn't my code that generated them. Also, I don't want to disturb EVE maintainers by pushing a PR that will fail on yetus later, so I use to run it locally before pushing my PRs, and I think this must be the normal workflow.... we should be able to easily see errors generated by our changes.... like we can clearly see when using, for instance, checkpatch.pl....

> I was about to suggest that blanks and codespell could be applied blindly, but not the other. But then I see that blanks in this PR wants to correct verbatim output of a shell command, which makes no sense (even if it doesn't result in a bug AFAIK). 

Well, these corrections that you mentioned are only replacing TABs (\t) by four spaces. Except by the mariadb.txt, these replacements were applied on documentation files only. IMHO is not that terrible wrong and bad replace a \t from a command output by four spaces in the documentation (it's more a matter of formatting than changing the contents). But I agree 100% that we shouldn't touch on the verbatim output, as long as yetus doesn't claim about it.

If codespell was limited to comments and markdown then it would be safe, but I've seen one case when it was commenting on a variable name ("ans" is apparently a mistyped "and" according to codespell) and I can see cases where renaming a variable can lead to subtle bugs (supposed the code has one variable called "answer" and another called "ans", and as a result of codespell the user renames "ans" to "answer" and not things are broken.)
> 
> So I think we need to be a lot more careful with doing bulk changes based on these tools. Maybe doing a PR with only blanks (and even in that case not doing them blindly), and a separate one based on codespell?
> 

I agree, so I will split this PR into multiple PRs, maybe for now only correcting markdown and yaml stuff, since it seems all reviewers claims are falling on blanks changes, what do you think @eriknordmark ?

> But in general we will have complains creep into existing code since the defaults for the tools sometimes changes when there are new releases of these tools.

Well, I think that if we are using these kind of tools, definitely our source code must complain with them. As an analogy, it's like Linux kernel project force patches to complain with checkpatch.pl and the kernel source code itself doesn't complain.... doesn't make sense... so for that, I see some possible solutions:

- Add exceptions to yetus when they are reasonable to be added
- Fix the code to complain 100% with yetus
- Get rid of yetus (and use another tool)

We don't need to fix all errors at once, but I think if we commit to put some efforts to fix these errors throughout the time it can work well. Please, let me know your thoughts, @eriknordmark ? @giggsoff ?

File: api/proto/info/info.proto
Comment (rene) [reply api/proto/info/info.proto:445]: Ok, will do it...

**Comment by rene** [comment]
Updates in this PR: Break fixes for yamllint plugin errors into a new PR (https://github.com/lf-edge/eve/pull/2833)

**Comment by rene** [comment]
Updates in this PR: Break fixes for markdownlint plugin errors into a new PR (https://github.com/lf-edge/eve/pull/2834)

Let's discuss the changes of hard tabs (\t) there. The changes in this PR (to fix blanks errors) are more sensible, since it touches on source code. I will review again according comments pointed by reviewers and update this PR.

File: pkg/pillar/rootfs/blink.sh
Comment (eriknordmark) [inline pkg/pillar/rootfs/blink.sh:20]: Odd that you lost the indentation for this block.

File: pkg/pillar/rootfs/sos.sh
Comment (eriknordmark) [inline pkg/pillar/rootfs/sos.sh:8]: Ditto.
Better to delete this blink.sh so it was never used.

File: pkg/pillar/rootfs/blink.sh
Comment (eriknordmark) [reply pkg/pillar/rootfs/blink.sh:20]: Better to delete this blink.sh since it was never used.

File: tools/getimgversion.sh
Comment (eriknordmark) [inline tools/getimgversion.sh:9]: Why do we loose the identation here?
FWIW this script is useful and used.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
I suspect pkg/kvm-tools/include/sys/sdt.h isn't something we wrote hence doesn't make sense fixing it. But I don't know the origin of that file.

Also, the tests/eden file(s) should be fixed in eden as well otherwise we'll get those space issue back when a new version is copied/vendored to the eve repo.

**Comment by rene** [comment]
> I suspect pkg/kvm-tools/include/sys/sdt.h isn't something we wrote hence doesn't make sense fixing it. But I don't know the origin of that file.
> 
> Also, the tests/eden file(s) should be fixed in eden as well otherwise we'll get those space issue back when a new version is copied/vendored to the eve repo.

Ok, so in this case I think would be better to add them as exceptions, as it is done for firmware files (pkg/fw), for example. I'm reworking this PR....

File: tools/getimgversion.sh
Comment (rene) [reply tools/getimgversion.sh:9]: Good catch, I have no idea why indentation got lost here, each \t should be replaced by four spaces... I will check....

Comment (rene) [reply tools/getimgversion.sh:9]: @eriknordmark , it turns out that only one tab (\t) was used in the original file, but it seems Github's code viewer is smart enough to put the right indentation... I will fix the indentation...

Comment (rene) [reply tools/getimgversion.sh:9]: actually if the viewer uses 8 spaces (instead of 4) for each \t then it should look indented, and that's a good example why tabs and spaces shouldn't be mixed in the source code....

**Comment by rene** [comment]
Updates in this PR:

- Break the PR into two commits: one that fixes blank errors and a second one that fixes only blank-eol errors
- Address reviewers comments
- Exclude third party code from yetus checks
- Plugin blanks is now passing (no errors)

File: .yetus/excludes.txt
Comment (eriknordmark) [inline .yetus/excludes.txt:12]: My exclude this? We want checks on the proto files so we can incrementally improve things and make new defintions better. But we are not doing changes to existing definitions.

File: pkg/pillar/rootfs/blink.sh
Comment (eriknordmark) [inline pkg/pillar/rootfs/blink.sh:7]: please let's delete blink.sh instead of fixing it. (that was a previous comment of mine that appears to have gotten lost)

File: pkg/pillar/rootfs/sos.sh
Comment (eriknordmark) [inline pkg/pillar/rootfs/sos.sh:6]: please let's delete sos.sh instead of fixing it. (that was a previous comment of mine that appears to have gotten lost)

File: pkg/pillar/rootfs/blink.sh
Comment (rene) [reply pkg/pillar/rootfs/blink.sh:7]: Sorry, I think I've missed. I will delete.

File: pkg/pillar/rootfs/sos.sh
Comment (rene) [reply pkg/pillar/rootfs/sos.sh:6]: I will delete.

File: .yetus/excludes.txt
Comment (rene) [reply .yetus/excludes.txt:12]: I've excluded because we are not checking the output (api/go)... I will enable the checks again...

**Comment by rene** [comment]
Updates in this PR:

- Remove unused scripts: sos.sh and blink.sh
- Bring back checks for api/proto
- Fix yetus blanks errors for api/proto and update generated files (into a separate commit)

File: .yetus/excludes.txt
Comment (eriknordmark) [inline .yetus/excludes.txt:9]: But we have code in pkg/kvm-tools (like the Dockerfile etc) which we do want to check.
Can't you exclude something more specific?

**Review by eriknordmark — APPROVED** [review]
My only remaining issue is the overly broad pkg/kvm-tools in the yetus excludes. Once that is addressed this can be merged.

Note also that you have conflicts on some files.

File: .yetus/excludes.txt
Comment (rene) [reply .yetus/excludes.txt:9]: Indeed, we can exclude only pkg/kvm-tools/include and pkg/kvm-tools/dtrace... will update the PR...

**Comment by rene** [comment]
> My only remaining issue is the overly broad pkg/kvm-tools in the yetus excludes. Once that is addressed this can be merged.
> 
> Note also that you have conflicts on some files.

Ok, I will rebase and update the PR.

**Pull request merged**
